### PR TITLE
refactor: adjust some logic in huawei adapter

### PR DESCRIPTION
**Pull request by chlins**

### CODE REVIEW

**Comment by steven-zou** [comment]
@ywk253100 @bitsf 

Can one of you double-check this PR? If everything is ok, I'll merge the PR

File: src/replication/adapter/huawei/huawei_adapter.go [hunk 1]
```diff
@@ -237,9 +216,30 @@ func AdapterFactory(registry *model.Registry) (adp.Adapter, error) {
...
+				UserAgent: adp.UserAgentReplication,
+			}}
+		authorizer modifier.Modifier
+	)
+	if registry.Credential != nil {
+		authorizer = auth.NewBasicAuthCredential(
+			registry.Credential.AccessKey,
+			registry.Credential.AccessSecret)
+		modifiers = append(modifiers, authorizer)
+	}
```
Comment (bysph) [inline src/replication/adapter/huawei/huawei_adapter.go]: 这个地方有点问题，华为的镜像仓库只有第一个接口使用Basic认证，后续的所有v2接口都使用第一个接口获取的jwt来认证鉴权（Authorization: Bearer格式），这个modifier会导致后续接口认证出错，从而导致镜像同步失败。

Comment (bysph) [reply src/replication/adapter/huawei/huawei_adapter.go]: image

Comment (bysph) [reply src/replication/adapter/huawei/huawei_adapter.go]: 2019-11-13T12:08:50Z [INFO] [/replication/transfer/image/transfer.go:95]: client for source registry [type: harbor, URL: http://core:8080, insecure: true] created
2019-11-13T12:08:50Z [INFO] [/replication/transfer/image/transfer.go:105]: client for destination registry [type: huawei-SWR, URL: https://100.95.181.176:5300, insecure: true] created
2019-11-13T12:08:50Z [INFO] [/replication/transfer/image/transfer.go:138]: copying library/nginx:[test](source registry) to sphtest/nginx:[test](destination registry)...
2019-11-13T12:08:50Z [INFO] [/replication/transfer/image/transfer.go:157]: copying library/nginx:test(source registry) to sphtest/nginx:test(destination registry)...
2019-11-13T12:08:50Z [INFO] [/replication/transfer/image/transfer.go:261]: pulling the manifest of image library/nginx:test ...
2019-11-13T12:08:51Z [INFO] [/replication/transfer/image/transfer.go:272]: the manifest of image library/nginx:test pulled
2019-11-13T12:08:51Z [ERROR] [/replication/transfer/image/transfer.go:320]: failed to check the existence of the manifest of image sphtest/nginx:test on the destination registry: [401][{"errors":[{"code":"DENIED","message":"Authenticate Error"}]}]
2019-11-13T12:08:51Z [ERROR] [/replication/transfer/image/transfer.go:143]: [401][{"errors":[{"code":"DENIED","message":"Authenticate Error"}]}]

Comment (chlins) [reply src/replication/adapter/huawei/huawei_adapter.go]: @bysph 感谢指出，我会fix一下

Comment (bysph) [reply src/replication/adapter/huawei/huawei_adapter.go]: 好的，我之前没有看到你的回复，另外提了个issue跟踪，#9874。

**Pull request merged**